protocols/purple/purple.c

```diff
@@ -1107,6 +1116,24 @@ static PurpleBlistUiOps bee_blist_uiops =
 	prplcb_blist_remove,       /* remove */
 };
 
+void purple_hint_chat_names(struct groupchat *gc, PurpleConversation *conv) {
+	char *hint = NULL;
+	if (conv->title != NULL) {
+		hint = conv->title;
+	}
+	if (set_getbool(&(gc->ic->acc->set), "use_matrix_alias") && purple_conversation_get_data(conv, "matrix_alias") != NULL) {
+		hint = (char *) purple_conversation_get_data(conv, "matrix_alias");
+	}
+	if (set_getbool(&(gc->ic->acc->set), "name_with_tag")) {
+		char tagged_name[128];
+		char *tag = set_getstr(&(gc->ic->acc->set), "tag");
+		g_snprintf(tagged_name, sizeof(tagged_name), "%s-%s", tag, hint);
+		imcb_chat_name_hint(gc, tagged_name);
+	}
+	else {
+		imcb_chat_name_hint(gc, conv->title);
```

Comment:
This should probably use hint here? If name_with_tag is off and use_matrix_alias is on, it wouldn't use the matrix alias